Mathlib/AlgebraicGeometry/EllipticCurve/Jacobian.lean

```diff
@@ -1568,57 +1625,227 @@ end Affine
+    f ((W'.baseChange A).toJacobian.dblU P) := by
+  rw [← RingHom.coe_coe, ← map_dblU, map_baseChange]
+
+lemma baseChange_dblZ (P : Fin 3 → A) : (W'.baseChange B).toJacobian.dblZ (f ∘ P) =
+    f ((W'.baseChange A).toJacobian.dblZ P) := by
+  rw [← RingHom.coe_coe, ← map_dblZ, map_baseChange]
+
+lemma baseChange_dblX (P : Fin 3 → A) : (W'.baseChange B).toJacobian.dblX (f ∘ P) =
+    f ((W'.baseChange A).toJacobian.dblX P) := by
+  rw [← RingHom.coe_coe, ← map_dblX, map_baseChange]
+
+lemma baseChange_negDblY (P : Fin 3 → A) : (W'.baseChange B).toJacobian.negDblY (f ∘ P) =
+    f ((W'.baseChange A).toJacobian.negDblY P) := by
+  rw [← RingHom.coe_coe, ← map_negDblY, map_baseChange]
+
+lemma baseChange_dblY (P : Fin 3 → A) : (W'.baseChange B).toJacobian.dblY (f ∘ P) =
+    f ((W'.baseChange A).toJacobian.dblY P) := by
+  rw [← RingHom.coe_coe, ← map_dblY, map_baseChange]
+
+lemma baseChange_dblXYZ (P : Fin 3 → A) : (W'.baseChange B).toJacobian.dblXYZ (f ∘ P) =
+    f ∘ (W'.baseChange A).toJacobian.dblXYZ P := by
+  rw [← RingHom.coe_coe, ← map_dblXYZ, map_baseChange]
+
+lemma baseChange_addX (P Q : Fin 3 → A) : (W'.baseChange B).toJacobian.addX (f ∘ P) (f ∘ Q) =
+    f ((W'.baseChange A).toJacobian.addX P Q) := by
+  rw [← RingHom.coe_coe, ← map_addX, map_baseChange]
+
+lemma baseChange_negAddY (P Q : Fin 3 → A) : (W'.baseChange B).toJacobian.negAddY (f ∘ P) (f ∘ Q) =
+    f ((W'.baseChange A).toJacobian.negAddY P Q) := by
+  rw [← RingHom.coe_coe, ← map_negAddY, map_baseChange]
+
+lemma baseChange_addY (P Q : Fin 3 → A) : (W'.baseChange B).toJacobian.addY (f ∘ P) (f ∘ Q) =
+    f ((W'.baseChange A).toJacobian.addY P Q) := by
+  rw [← RingHom.coe_coe, ← map_addY, map_baseChange]
+
+lemma baseChange_addXYZ (P Q : Fin 3 → A) : (W'.baseChange B).toJacobian.addXYZ (f ∘ P) (f ∘ Q) =
+    f ∘ (W'.baseChange A).toJacobian.addXYZ P Q := by
+  rw [← RingHom.coe_coe, ← map_addXYZ, map_baseChange]
+
+variable [Algebra R F] [Algebra S F] [IsScalarTower R S F] {K : Type v} [Field K] [Algebra R K]
+  [Algebra S K] [IsScalarTower R S K] (f : F →ₐ[S] K)
+
+lemma baseChange_add {P Q : Fin 3 → F} (hP : (W'.baseChange F).toJacobian.Nonsingular P)
+    (hQ : (W'.baseChange F).toJacobian.Nonsingular Q) :
+    (W'.baseChange K).toJacobian.add (f ∘ P) (f ∘ Q) =
+      f ∘ (W'.baseChange F).toJacobian.add P Q := by
+  rw [← RingHom.coe_coe, ← WeierstrassCurve.Jacobian.map_add f hP hQ (K := K), map_baseChange]
+
+end BaseChange
+
 end WeierstrassCurve.Jacobian
 
 /-- An abbreviation for `WeierstrassCurve.Jacobian.Point.fromAffine` for dot notation. -/
-abbrev WeierstrassCurve.Affine.Point.toJacobian {R : Type u} [CommRing R]
-    [Nontrivial R] {W : Affine R} (P : W.Point) : W.toJacobian.Point :=
+abbrev WeierstrassCurve.Affine.Point.toJacobian {R : Type u} [CommRing R] [Nontrivial R]
+    {W : Affine R} (P : W.Point) : W.toJacobian.Point :=
   Jacobian.Point.fromAffine P
 
-set_option linter.style.longFile 1700
+set_option linter.style.longFile 1900
```

Comment:
It would be nice if you could split it up in a later PR.

Reply:
This will be done very soon!